py/torch_tensorrt/dynamo/_compiler.py

```diff
@@ -197,6 +201,20 @@ def compile(
             "\nThis feature is unimplemented in Torch-TRT Dynamo currently."
         )
 
+    if use_strong_types:
+        if len(enabled_precisions) != 1 or not any(
+            x in enabled_precisions for x in {torch.float32, dtype.f32}
+        ):
+            raise AssertionError(
+                f"When use_strong_types is enabled, only torch.float32 is allowed in the enabled_precisions but found {enabled_precisions}"
+            )
+
+    if use_fp32_acc:
+        logger.debug(
```

Comment:
Should there be deeper input validation here? like checking if fp16 is in enabled precisions and throwing a warning if fp16 is not in the set but use_fp32_acc is True?  The second line should be part of the docs.

Reply:
-use_fp32_acc - replicate pytorch fp16 matmul behavior 
- use_strong typing - some layers in llm moels are explicity casted to fp32 (replicate this in TRT) 
- enabled_precisions - should be not have any other than FP32 

- amp example to demo strong typing (userguide/ tutorial) 
- use_fp32_acc + use_strong_typing (llm examples demo)